cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,367 @@
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+The XML representation used is deliberately verbose in favor of readability
+and deterministic processing.
+
+Preservation of type information for CloudEvent attributes is supported allowing
+custom extensions to be communicated without type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+The namespace `http://cloudevents.io/xmlformat/V1` MUST be used.
+
+When namespace prefixes are used a prefix of `ce` is preferred but MUST NOT
+be expected from an XML document processing perspective.
+
+An XML document preamble SHOULD be included to ensure deterministic processing.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to a set of CloudEvent specific
+type designators as follows :
+
+| CloudEvents Type  |  XML Format Type | XML Schema Type | Notes |
+| :-----------------| :----------------| :--------------- | :---- |
+| Boolean       | ce:boolean | [xs:boolean][xml-primitives] | `true` or `false` |
+| Integer       | ce:int | [xs:int][xml-primitives] | |
+| String        | ce:string | [xs:string][xml-primitives] | |
+| Binary        | ce:binary | [xs:base64Binary][xml-primitives] |  |
+| URI           | ce:uri | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | ce:uriRef | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | ce:timestamp | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each CloudEvent context attribute is represented as an XML element whose local
+name exactly matches that of the attribute.
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes MUST be decorated with the appropriate CloudEvent format
+designators using using the `xsi:type` XML attribute, this allows them to be exchanged
+without loss of type information.
+
+REQUIRED and OPTIONAL context attributes SHOULD NOT be decorated with an `xsi:type`.
+
+No other XML element attributes are expected, if present they MUST be ignored during
+processing.
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T14:30:22-08:00</time>
+    ..
+    <myextension xsi:type="ce:string">my extension value</myextension>
+    <myboolean xsi:type="ce:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The data portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. A `data` element MUST be used to
+encapsulate the payload.
+
+An `xsi:type` is used to discrimate the payload type and MUST be present.
+
+The following data representations are supported:
+
+### 3.1 Binary Data
+
+Binary data MUST be carried in an element with an defined type of `ce:binary`.
+
+Example:
+
+``` xml
+<data xsi:type="ce:binary">.........</data>
+```
+
+### 3.2 Text Data
+
+Text MUST be carried in an element with an defined type of `ce:string`.
```

Comment:
Do you think we need to mention that it needs to be xml-escaped?

Reply:
I would hope not, but i can add that clarification - it does sound a bit redundant though as if the escaping hasn't happened then the overall XML document wouldn't be valid and wouldn't parse.

Reply:
I agree with Jem on this one. I think "it should be valid XML" can be assumed. (And indeed I've taken that approach to testing my implementation: the tests should be of the formatter code, not the XML parser.)